pyg_lib/csrc/random/cpu/randint_engine.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <torch/torch.h>
+
+#include <limits.h>
+
+namespace pyg {
+namespace random {
+
+const int RAND_PREFETCH_THRESHOLD = 128;
+
+// Use torch::randint to generate 64-bit random numbers
+const int RAND_PREFETCH_BITS = 64;
+
+/**
+ * Amortized O(1) uniform random within a range.
+ *
+ * Reduced torch random overheads by prefetching large random bits in advance.
+ * Prefetched random bits are consumed on demand.
+ *
+ */
+class PrefetchedRandint {
+ public:
+  PrefetchedRandint()
+      : PrefetchedRandint(RAND_PREFETCH_THRESHOLD, RAND_PREFETCH_BITS) {}
+  PrefetchedRandint(int size, int bits) : size_(size), bits_(bits) {
```

Comment:
I think we don't need `size_(size), bits_(bits)` since they were initialized in `prefetch` function.